docs/usage/CUSTOMER_SYNC.md

```diff
@@ -0,0 +1,106 @@
+````
+
+[More information about Sync Options](SYNC_OPTIONS.md).
+
+#### Running the sync
+When all prerequisites are fulfilled, follow those steps to run the sync:
+
+````java
+// instantiating a cart discount sync
+final CustomerSync customerSync = new CustomerSync(customerSyncOptions);
+
+// execute the sync on your list of customers
+CompletionStage<CustomerSyncStatistics> syncStatisticsStage = customerSync.sync(customerDrafts);
+````
+The result of the completing the `syncStatisticsStage` in the previous code snippet contains a `CustomerSyncStatistics`
+which contains all the stats of the sync process; which includes a report message, the total number of updated, created,
+failed, processed cart discounts, and the processing time of the last sync batch in different time units and in a
+human-readable format.
+
+````java
+final CustomerSyncStatistics stats = syncStatisticsStage.toCompletebleFuture().join();
+stats.getReportMessage();
+/*"Summary: 100 customers were processed in total (11 created, 87 updated, 2 failed to sync)."*/
+````
+
+__Note__ The statistics object contains the processing time of the last batch only. This is due to two reasons:
+
+ 1. The sync processing time should not take into account the time between supplying batches to the sync.
+ 2. It is not known by the sync which batch is going to be the last one supplied.
+
+#### More examples of how to use the sync
+
+ [Sync from an external source](https://github.com/commercetools/commercetools-sync-java/tree/master/src/integration-test/java/com/commercetools/sync/integration/externalsource/customers/CustomerSyncIT.java).
+
+*Make sure to read the [Important Usage Tips](IMPORTANT_USAGE_TIPS.md) for optimal performance.*
+
+### Build all update actions
+A utility method provided by the library to compare a `Customer` to a new `CustomerDraft`. The results are collected in a list of customer update actions.
+```java
+List<UpdateAction<Customer>> updateActions = CustomerSyncUtils.buildActions(customer, customerDraft, customerSyncOptions);
+```
+
+### Build particular update action(s)
+The library provides utility methods to compare specific fields of a `Customer` and a new `CustomerDraft`, and builds the update action(s) as a result.
+One example is the `buildChangeEmailUpdateAction` which compare email addresses:
+````java
+Optional<UpdateAction<Customer>> updateAction = CustomerUpdateActionUtils.buildChangeEmailAction(oldCustomer, customerDraft);
+````
+
+More examples for particular update actions can be found in the test scenarios for [CustomerUpdateActionUtils](https://github.com/commercetools/commercetools-sync-java/tree/master/src/test/java/com/commercetools/sync/customers/utils/CustomerUpdateActionUtilsTest.java)
+and [AddressUpdateActionUtils](https://github.com/commercetools/commercetools-sync-java/tree/master/src/test/java/com/commercetools/sync/customers/utils/AddressUpdateActionUtilsTest.java).
+
+
+##Caveats
+Customer passwords are not being synced.
```

Comment:
Here this is only for update cases, not for the creation of a new customer, so can you describe it in more clear way.

Reply:
What happens if a new customer is created? It is created with all the information that could be synced but without a password?

Reply:
78163328